### ISSUE TITLE
How to generate the textual descriptions for my own dataset?

### ISSUE THREAD

**Issue by Crane-YU** [sidward14/Style-AttnGAN#7]
Hi @sidward14,

Thx for sharing your work. While I was reading the original paper, I was curious about how the texts are generated for the CUB dataset. Is there a fixed pattern I can follow or is it generated based on a fixed grammar?

**Comment by sidward14** [comment]
No need to follow a fixed pattern or grammar. You can vary how you describe each image, though doing so will make the problem more difficult for the model to learn. I'm curious how well it will do. If using very long descriptions, perhaps using the transformer (instead of the default bilstm) would be better.

**Comment by Crane-YU** [comment]
@sidward14 Thank you. I've read through the work and found that the authors manually generated the textual descriptions. I am wondering if I use some existing caption-generation network to generate the texts, will this introduce distribution gap and thus harm the model performance?

**Comment by sidward14** [comment]
If the generated captions are consistent with the images, then the model should be able to learn them. Distribution gap between what?

**Comment by Crane-YU** [comment]
By distribution gap I mean the textual descriptions. As I'm looking for fine-tuning, the textual descriptions I generated via caption-generator might have a different distribution (I suppose) than the human-generated textual descriptions. Will this introduce some sort of diatribution gap?

**Comment by sidward14** [comment]
It should be fine. There will be some differences in model-generated vs human-generated distributions. Biggest one is that the distribution for model-generated captions is likely simpler. But if your caltion-generator is decent, it shouldn't be a huge problem.

For your training set: the larger the distribution gap, the more training data you'll need and the more epochs you'll need to train for fine-tuning.

For your validation set: If during inference time (when you're actually using this model), you want to use captions from humans, then try to make your validation set captions human-generated.

**Comment by Crane-YU** [comment]
Got it. Thank you.
